### PR TITLE
add grafana e2e

### PR DESCRIPTION
**Pull request by ldpliu**
https://issues.redhat.com/browse/ACM-6245

### CODE REVIEW

File: operator/pkg/controllers/hubofhubs/integration_test.go [hunk 1]
```diff
@@ -943,12 +943,20 @@ var _ = Describe("MulticlusterGlobalHub controller", Ordered, func() {
...
+			Eventually(func() error {
+				mghReconciler.MiddlewareConfig.PgConnection = nil
+				mghReconciler.MiddlewareConfig.KafkaConnection = nil
+
+				mghReconciler.ReconcileMiddleware(ctx, mcgh)
+				if mghReconciler.MiddlewareConfig.PgConnection != nil {
+					return fmt.Errorf("mghReconciler.MiddlewareConfig.PgConnection should be nil")
+				}
+				if mghReconciler.MiddlewareConfig.KafkaConnection != nil {
+					return fmt.Errorf("mghReconciler.MiddlewareConfig.KafkaConnection should be nil")
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/integration_test.go:955]: why need to wrap in `Eventually`. I think it just runs once. Am I missing sth?

File: test/pkg/e2e/grafana_test.go [hunk 2]
```diff
@@ -258,3 +266,120 @@ func sectionCount(a []byte) int {
...
+			"fbf66c88-ba14-4553-90b7-55ea5870faab", //overview
+			"9bb3bee6a17e47f9a231f6d77f2408fa",     //policy group compliance overview
+			"5a3a577af7894943aa6e7ca8408502fb",     //what's changed cluster
+			"5a3a577af7894943aa6e7ca8408502fa",     //what's changed policies
+		}
+		for _, uid := range dashboardUids {
+			Eventually(func() error {
+				_, err := getGrafanaResource(ctx, "dashboards/uid/"+uid)
+				return err
+			}, 2*time.Minute, 10*time.Second).ShouldNot(HaveOccurred())
```
Comment (clyang82) [inline test/pkg/e2e/grafana_test.go:301]: will it spend too long?

File: test/pkg/e2e/grafana_test.go [hunk 3]
```diff
@@ -258,3 +266,120 @@ func sectionCount(a []byte) int {
...
+	})
+	if err != nil {
+		return 0, err
+	}
+	klog.Infof("Get grafana path:%v", path)
+	if len(poList.Items) == 0 {
+		return 0, fmt.Errorf("Can not get grafana pod")
+	}
+
+	url := "http://" + poList.Items[0].Status.PodIP + ":3001/api/" + path
```
Comment (clyang82) [inline test/pkg/e2e/grafana_test.go:326]: Can pod ip be accessed outside?

**Review by clyang82 — CHANGES_REQUESTED** [review]
Can you confirm the new added cases are performed? Do we need to add `e2e-tests-grafana-resources` in Makefile ?

File: operator/pkg/controllers/hubofhubs/integration_test.go (hunk 1 above)
Comment (ldpliu) [reply operator/pkg/controllers/hubofhubs/integration_test.go:955]: This part always failed with some conflict error.
like: 
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/stolostron_multicluster-global-hub/699/pull-ci-stolostron-multicluster-global-hub-main-test-unit/1716721252795682816#1:build-log.txt%3A1574
and 
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/stolostron_multicluster-global-hub/699/pull-ci-stolostron-multicluster-global-hub-main-test-unit/1716724914012033024#1:build-log.txt%3A1584

File: test/pkg/e2e/grafana_test.go (hunk 2 above)
Comment (ldpliu) [reply test/pkg/e2e/grafana_test.go:300]: In my test, it response very quickly.

File: test/pkg/e2e/grafana_test.go (hunk 3 above)
Comment (ldpliu) [reply test/pkg/e2e/grafana_test.go:322]: Changed using kubectl exec to access the api

File: test/pkg/e2e/grafana_test.go (hunk 2 above)
Comment (clyang82) [reply test/pkg/e2e/grafana_test.go:300]: that is my understand. why the interval is 10 seconds? maybe update to 1 second.

File: test/pkg/e2e/grafana_test.go [hunk 4]
```diff
@@ -258,3 +263,101 @@ func sectionCount(a []byte) int {
...
+			if err != nil {
+				klog.Errorf("Get grafana resource error:%v", err)
+				return err
+			}
+			if alertCount != 5 {
+				klog.Errorf("get alert count:%v", alertCount)
+				return fmt.Errorf("Expect alert count is 5, bug got:%v", alertCount)
+			}
+			return nil
+		}, 2*time.Minute, 10*time.Second).ShouldNot(HaveOccurred())
```
Comment (clyang82) [inline test/pkg/e2e/grafana_test.go:281]: update the 1 second to reduce the possible waiting time.

**Review by clyang82 — APPROVED** [review]
/lgtm

**Pull request merged**